File: client-core/src/main/java/Anon/client/rulegeneration/core/resource/Flow.java
package Anon.client.rulegeneration.core.resource;

/**
 * Representation of attack flows for Anon demo
 *
 */

public class Flow {
    public String source;
    public String destination;
    public int protocol;
    public int bytes;
    public int packets;


    public Flow() {
        this.source = "";
        this.destination = "";
        this.protocol = 0;
        this.bytes = 0;
        this.packets = 0;
    }

    public Flow(IP source) {
        this.source = source.toString();
        this.destination = "";
        this.protocol = 0;
        this.bytes = 0;
        this.packets = 0;
    }
}

File: controller-core/src/main/kotlin/Anon/controller/core/traffic/Flow.kt
package Anon.controller.core.traffic

import Anon.controller.core.topology.AS
import Anon.controller.core.topology.AsTopology

/**
 * Representation of attack flows in a DDoS attack.
 */
data class Flow (
        var ip: IP,
        var bitRate: Int = 1,
        var filteredAt: AS? = null,
        val timeStamp: Int = 0
)

